src/paho/mqtt/client.py

```diff
@@ -1734,7 +1764,10 @@ def _packet_read(self):
                     print(err)
                     return 1
                 else:
-                    byte, = struct.unpack("!B", byte)
+                    if len(byte) == 0:
+                        return 1
+                    byte = struct.unpack("!B", byte)
```

Comment:
Yes, because the new unpacking was added by #81 merged in develop and not fixes.
Therefor your merge is rolling back a part of #81.